### PR TITLE
Chemistry codex - VueUI edition

### PR DESCRIPTION
**Pull request by Karolis2011**
#9388 just vueui edition

Mostly contains fuzzy search what should make search experence nice and responsive. Data generation assumes rections do not change.
image

Need feedback on program access restrctions and size

### CODE REVIEW

**Comment by Karolis2011** [comment]
!review

**Comment by Geevies** [comment]
oh also, can you add a button that hides the descriptions like mine did? and reduce the spacing between the h elements and the text, it's a bit big. the description font size can be lowered

File: code/modules/modular_computers/file_system/programs/research/chemistry_codex.dm [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	var/headerdata = get_header_data(data["_PC"])
+	if(headerdata)
+		data["_PC"] = headerdata
+		. = data
+	
+	// Here goes listification
+	if(data["reactions"] == null)
+		. = data
+		data["reactions"] = list()
+		for(var/chem_path in SSchemistry.chemical_reactions_clean)
```
Comment (NonQueueingMatt) [inline code/modules/modular_computers/file_system/programs/research/chemistry_codex.dm:36]: This stuff should just be built at SS init and then fetched later.

**Comment by Chada-1** [comment]
Thank you Karolis, this was very nice

**Comment by Karolis2011** [comment]
Also please help me to decide on: what devices should be able to run it?
What access is needed?
How big should program be?

**Comment by NonQueueingMatt** [comment]
IMO make it runnable on wristbounds. Should be decently big though.

**Comment by Chada-1** [comment]
Should probably hold off on merging until Karolis is sure it's ready, I didn't realize stuff was still being requested when I approved. Sorry :eyes:

**Comment by Karolis2011** [comment]
If size and access seems appropriate, then feel free to merge.

**Comment by skull132** [comment]
For posterity: does this handle secret chems properly?

**Comment by Karolis2011** [comment]
Re to Skull: Data is built before screrect chems are loaded, so it should not contain them. See https://github.com/Aurorastation/Aurora.3/blob/99d67a677d5e570e2bb56426976f645d3fe120f2/code/controllers/subsystems/chemistry.dm#L85

**Pull request merged**